src/utils/ragEngine.ts

```diff
@@ -0,0 +1,107 @@
+import { App } from 'obsidian'
+
+import { QueryProgressState } from '../components/chat-view/QueryProgress'
+import { SelectVector } from '../db/schema'
+import { EmbeddingModel } from '../types/embedding'
+import { SmartCopilotSettings } from '../types/settings'
+
+import { VectorDbManager } from './vector-db/manager'
+
+export class RAGEngine {
+  private settings: SmartCopilotSettings
+  private vectorDbManager: VectorDbManager
+  private embeddingModel: EmbeddingModel | null = null
+
+  constructor(app: App, settings: SmartCopilotSettings) {
+    this.settings = settings
+    this.vectorDbManager = new VectorDbManager(app)
+  }
+
+  async initialize(embeddingModel: EmbeddingModel): Promise<void> {
+    await this.vectorDbManager.initialize()
+    this.setEmbeddingModel(embeddingModel)
+  }
```

Comment:
Creating a `static async create()` function instead of creating an instance and calling initialize every time would be a better approach. This way, we can perform the necessary async operations within this function before creating the instance. This approach would also help improve the repetitive code in all functions here, such as:

```
if (!this.embeddingModel) {
  throw new Error('Embedding model is not set');
}
```

By using a static create method, we can ensure that the embedding model is properly set during object creation, eliminating the need for these checks in every function.

This is also applicable to other classes like `VectorDbManager` or `VectorDbRepository`.

Reply:
like this:
```
static async create(app: App): Promise<VectorDbManager> {
    const manager = new VectorDbManager(app)
    await ~~
    return manager
}
```